### PR TITLE
[Rom e2e] bootstrap watchdog disable

### PR DESCRIPTION
**Pull request by engdoreis**
Add test `e2e_bootstrap_watchdog_disable`

### CODE REVIEW

**Review by alphan — COMMENTED** [review]
Thanks for taking a stab at this @engdoreis! I think you are on the right track. Please see my comments below.

- [x] I don't think this test needs a on-target component, i.e. you can probably remove `rom_e2e_bootstrap_watchdog_disable_test.c` and related changes.
- [x] Can you try adding this testpoint to `sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs` instead of a separate rust program?
- [ ] You can use the conditions described in #14455 to determine success (`READ_SFDP`) and failure (any UART output).

**Comment by engdoreis** [comment]
@alphan thanks for the feedback, I've done the first two items, but I didn't get the last one. Could give a second look?

File: sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs [hunk 1]
```diff
@@ -585,13 +630,13 @@ fn main() -> Result<()> {
...
-    execute_test!(test_bootstrap_phase1_page_program, &opts, &transport);
-    for erase_cmd in [SpiFlash::SECTOR_ERASE, SpiFlash::CHIP_ERASE] {
-        execute_test!(test_bootstrap_phase1_erase, &opts, &transport, erase_cmd);
+        ] {
+            execute_test!(test_bootstrap_shutdown, &opts, &transport, cmd, bfv);
+        }
+        execute_test!(test_bootstrap_phase1_reset, &opts, &transport);
+        execute_test!(test_bootstrap_phase1_page_program, &opts, &transport);
+        for erase_cmd in [SpiFlash::SECTOR_ERASE, SpiFlash::CHIP_ERASE] {
+            execute_test!(test_bootstrap_phase1_erase, &opts, &transport, erase_cmd);
```
Comment (alphan) [inline sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs:639]: nit: can you take a look at the indentation and whitespace changes here?

File: sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs [hunk 2]
```diff
@@ -570,11 +570,56 @@ fn test_bootstrap_phase2_read(opts: &Opts, transport: &TransportWrapper) -> Resu
...
+
+    // Verify that there was no output from UART.
+    console.exit_success = Some(Regex::new(r".+")?);
+    console.exit_failure = Some(Regex::new(r".+")?);
+    let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
+    if result != ExitStatus::Timeout {
+        bail!("FAIL: {:?}", result);
+    };
+
+    Ok(())
```
Comment (alphan) [inline sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs:615]: I think we want something like the following. Please review and verify. Thanks!

```suggestion
    let _bs = BootstrapTest::start(transport, opts.init.bootstrap.options.reset_delay)?;
    let spi = transport.spi("0")?;
    let uart = transport.uart("0")?;
    let mut console = UartConsole {
        timeout: Some(Duration::new(2, 0)),
        exit_success: Some(Regex::new(r".+")?),
        exit_failure: Some(Regex::new(r".+")?),
        ..Default::default()
    };
    
    // Verify that the chip responds to `READ_SFDP` (`0x5a`). 
    let sfdp_ = SpiFlash::read_sfdp(&*spi)?;
    // Release bootstrap pin strapping and verify that we don't receive
    // anything over UART until the console times out.
    uart.clear_rx_buffer()?;
    transport.remove_pin_strapping("ROM_BOOTSTRAP")?;
    let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
    if result != ExitStatus::Timeout {
        bail!("FAIL: {:?}", result);
    };
    // Verify that the chip is still in bootstrap.
    let sfdp_ = SpiFlash::read_sfdp(&*spi)?;
    
    Ok(())
```

**Comment by alphan** [comment]
> @alphan thanks for the feedback, I've done the first two items, but I didn't get the last one. Could give a second look?

Thanks @engdoreis! Done, I also took the liberty to edit the test point to check off the last item. PTAL

File: sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs (hunk 1 above)
Comment (engdoreis) [reply sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs:639]: Sorry, fixed.

**Review by alphan — APPROVED** [review]
Thanks @engdoreis, PTAL the CI errors.

**Comment by engdoreis** [comment]
I tested the condition where the chip resets by manually pressing the `USR-RESET` button in the CW310 and the test failed as expected.

**Comment by alphan** [comment]
> I tested the condition where the chip resets by manually pressing the `USR-RESET` button in the CW310 and the test failed as expected.

Thanks @engdoreis!

**Pull request merged**